src/daemon/devnode.c

```diff
@@ -297,12 +327,12 @@ int rmdevpath(usbdevice* kb){
     euid_guard_start;
     int index = INDEX_OF(kb, keyboard);
     if(kb->infifo != 0){
-        int fd = kb->infifo - 1;
+        FILE* fd = fdopen(kb->infifo - 1, "a");
 #ifdef OS_MAC
-        fcntl(fd, F_SETFL, O_RDWR | O_NONBLOCK); // hack to prevent the following hack from blocking if the GUI was running
+        fcntl(kb->infifo - 1, F_SETFL, O_RDWR | O_NONBLOCK); // hack to prevent the following hack from blocking if the GUI was running
 #endif
-        write(fd, "\n", 1); // hack to prevent the FIFO thread from perma-blocking
-        close(fd);
+        fputc('\n', fd); //hack to prevent the FIFO thread from perma-blocking
```

Comment:
This is suspicious to me. The reason the `write` call was here is because it bypasses libc buffering; while it should get flushed with the `fclose`, I'm unsure about changing this code for change's sake.